test/api/handlers_test.go

```diff
@@ -0,0 +1,813 @@
+
+	"github.com/NebulousLabs/skynet-accounts/database"
+	"github.com/NebulousLabs/skynet-accounts/email"
+	"github.com/NebulousLabs/skynet-accounts/skynet"
+	"github.com/NebulousLabs/skynet-accounts/test"
+	"gitlab.com/NebulousLabs/fastrand"
+	"gitlab.com/SkynetLabs/skyd/skymodules"
+	"go.mongodb.org/mongo-driver/bson"
+	"go.mongodb.org/mongo-driver/mongo/options"
+	"go.sia.tech/siad/crypto"
+)
+
+const (
+	// Helper consts for checking the returned errors.
+	badRequest   = "400 Bad Request"
+	unauthorized = "401 Unauthorized"
+)
+
+type subtest struct {
+	Name string
+	Test func(t *testing.T, at *test.AccountsTester)
+}
+
+// TestHandlers is a meta test that sets up a test instance of accounts and runs
+// a suite of tests that ensure all handlers behave as expected.
+func TestHandlers(t *testing.T) {
+	at, err := test.NewAccountsTester()
+	if err != nil {
+		t.Fatal(err)
+	}
+	defer at.Shutdown()
+
+	// Specify subtests to run
+	tests := []subtest{
+		// GET /health
+		{Name: "health", Test: testHandlerHealthGET},
+		// POST /user, POST /login
+		{Name: "userCreate", Test: testHandlerUserPOST},
+		// POST /login, POST /logout, POST /user
+		{Name: "loginLogout", Test: testHandlerLoginPOST},
+		// PUT /user
+		{Name: "userEdit", Test: testUserPUT},
+		// GET /user/limits
+		{Name: "userLimits", Test: testUserLimits},
+		// DELETE /user/uploads/:skylink, GET /user/uploads
+		{Name: "userDeleteUploads", Test: testUserUploadsDELETE},
+		// GET /user/confirm, POST /user/reconfirm
+		{Name: "userConfirmReconfirmEmail", Test: testUserConfirmReconfirmEmailGET},
+		// GET /user/recover, POST /user/recover, POST /login
+		{Name: "userAccountRecovery", Test: testUserAccountRecovery},
+		// POST /track/upload/:skylink, POST /track/download/:skylink, POST /track/registry/read, POST /track/registry/write, GET /user/stats
+		{Name: "standardTrackingFlow", Test: testTrackingAndStats},
+		// POST /user, POST /login, PUT /user, GET /user, POST /logout
+		{Name: "standardUserFlow", Test: testUserFlow},
+	}
+
+	// Run subtests
+	for _, tt := range tests {
+		t.Run(tt.Name, func(t *testing.T) {
+			tt.Test(t, at)
+		})
+	}
+}
+
+// testHandlerHealthGET tests the /health handler.
+func testHandlerHealthGET(t *testing.T, at *test.AccountsTester) {
+	_, b, err := at.Get("/health", nil)
+	if err != nil {
+		t.Fatal(err)
+	}
+	status := struct {
+		DBAlive bool `json:"dbAlive"`
+	}{}
+	err = json.Unmarshal(b, &status)
+	if err != nil {
+		t.Fatal("Failed to unmarshal service's response: ", err)
+	}
+	// DBAlive should never be false because if we couldn't reach the DB, we
+	// wouldn't have made it this far in the test.
+	if !status.DBAlive {
+		t.Fatal("DB down.")
+	}
+}
+
+// testHandlerUserPOST tests user creation and login.
+func testHandlerUserPOST(t *testing.T, at *test.AccountsTester) {
+	// Use the test's name as an email-compatible identifier.
+	name := strings.ReplaceAll(t.Name(), "/", "_")
+	email := name + "@siasky.net"
+	params := map[string]string{
```

Comment:
I'd build the map inside of `Post` and have a `at.CreateUserPost` method which accepts an email and password.